### DISCUSSION TITLE
Resolving IDs with mutation conventions

### DISCUSSION THREAD

**Discussion by pckv** [ChilliCream/hotchocolate#4641] [Q&A] [answered]
Hello!

I'm trying to deserialize an ID in a mutation while using the new `UseMutationConvention` attribute, but it's not being resolved.

Here is the mutation:
```cs
[UseMutationConvention]
public string ReturnId([ID] int id) => id.ToString();
```

And the result:
```graphql
{
  "errors": [
    {
      "message": "Unable to convert the value of the argument `id` to `System.Int32`. Check if the requested type is correct or register a custom type converter.",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "returnId"
      ],
      "extensions": {
        "fieldName": "returnId",
        "argumentName": "id",
        "requestedType": "System.Int32"
      }
    }
  ]
}
```

When I remove the `[UseMutationConvention]` attribute or create the input manually like shown below it works.

```cs
public record ReturnIdInput([property: ID] int id);
public string ReturnId(ReturnIdInput input) => input.id.ToString();
```

Am I missing something? Looking at the docs for [Input and Payload](https://chillicream.com/docs/hotchocolate/defining-a-schema/mutations/#input-and-payload) it seems like this should be supported.

Also I want to note how I need to declare the ID as `[property: ID]` in the record. Before v12 `[ID]` worked just fine. Is there any reason for this?

**Comment by tobias-tengler** [comment, marked as answer]
A fix is already in the works in #4639 :)

About the property syntax in records: It's the official way to annotate the resulting properties of these "constructor arguments". The solution of v11 was unmaintainable and led to many bugs, as mentioned in the [migration guide](https://chillicream.com/docs/hotchocolate/api-reference/migrate-from-11-to-12/#records).

**Reply by pckv** [reply]
Ah, fantastic! Thanks a lot :)
